lib/rspec/mocks/instance_method_stasher.rb

```diff
@@ -128,7 +128,7 @@ def method_owned_by_klass?
         # Hence, we verify that the owner actually has the method defined.
         # If the given owner does not have the method defined, we assume
         # that the method is actually owned by @klass.
-        owner == @klass || !(method_defined_on_klass?(owner))
+        owner == @klass || owner == @object || !(method_defined_on_klass?(owner))
```

Comment:
So this looks like an extension of the bug reported on line #112, would you mind just confirming what owner looks like in your example, and then elaborate with a comment for future reference.